packages/wix-ui-codemod/src/wix-style-react/migrate-wsr10.ts

```diff
@@ -11,19 +16,126 @@ type RemoveInterface = {
   value?: string;
 };
 
+const propertiesUpdates = {
+  Input: {
+    rename: [
+      {
+        from: { prop: 'size', value: 'normal' },
+        to: { prop: 'size', value: 'medium' },
+      },
+      {
+        from: { prop: 'roundInput' },
+        to: { prop: 'border', value: 'round' },
+      },
+    ],
+  },
+  InputArea: {
+    rename: [
+      {
+        from: { prop: 'size', value: 'normal' },
+        to: { prop: 'size', value: 'medium' },
+      },
+    ],
+  },
+  MarketingLayout: {
+    remove: [
+      {
+        prop: 'size',
+        value: 'large',
+      },
+    ],
+  },
+  RadioGroup: {
+    remove: [
+      {
+        prop: 'lineHeight',
+      },
+    ],
+  },
+  AutoComplete: {
+    rename: [
+      {
+        from: { prop: 'roundInput' },
+        to: { prop: 'border', value: 'round' },
+      },
+    ],
+  },
+  Dropdown: {
+    rename: [
+      {
+        from: { prop: 'roundInput' },
+        to: { prop: 'border', value: 'round' },
+      },
+    ],
+  },
+  MultiSelect: {
+    rename: [
+      {
+        from: { prop: 'roundInput' },
+        to: { prop: 'border', value: 'round' },
+      },
+    ],
+  },
+  MultiSelectCheckbox: {
+    rename: [
+      {
+        from: { prop: 'roundInput' },
+        to: { prop: 'border', value: 'round' },
+      },
+    ],
+  },
+  NumberInput: {
+    rename: [
+      {
+        from: { prop: 'roundInput' },
+        to: { prop: 'border', value: 'round' },
+      },
+    ],
+  },
+  Search: {
+    rename: [
+      {
+        from: { prop: 'roundInput' },
+        to: { prop: 'border', value: 'round' },
+      },
+    ],
+  },
+};
+
+const renamedComponents = [
+  {
+    originalName: 'TimeInputNext',
+    newName: 'TimeInput',
+  },
+  {
+    originalName: 'SortableList',
+    newName: 'SortableListBase',
+  },
+  {
+    originalName: 'StyledNestableList',
+    newName: 'NestableList',
+  },
+];
+
 const transform: Transform = (file, api) => {
   const j = api.jscodeshift;
   const root = j(file.source);
 
-  const findOpeningTag = (name: string) =>
-    root.find(j.JSXOpeningElement, { name: { name } });
+  const findWSRComponentImport = (name: string) =>
+    root
+      .find(j.ImportDeclaration, {
+        source: {
+          value: 'wix-style-react',
+        },
+      })
+      .find(j.ImportSpecifier, {
+        imported: { name },
+      });
 
-  const findJSXElementThatContains = (name: string) =>
-    root.find(j.JSXElement, {
-      openingElement: {
-        name: { name },
-      },
-    });
+  const findOpeningTagPaths = (name: string) =>
+    findWSRComponentImport(name).length
+      ? root.find(j.JSXOpeningElement, { name: { name } })
```

Comment:
```suggestion
      root.find(j.JSXOpeningElement, { name: { name } }).filter(path => findWSRComponentImport(name).length)
```

maybe something like this to secure ourselves from accidentally changing user components that have the same name as some wsr component, but is not imported from wsr?

for example, `<Input/>` in the following case should be unchanged

```diff
import { Input } from 'wix-ui-tpa/dist/src/components/Input';
-import { TimeInputNext } from 'wix-style-react';
+import { TimeInput } from 'wix-style-react';

// ...

<Input size="normal"/>
-<TimeInputNext />
+<TimeInput />
```